src/util/logging.py

```diff
@@ -17,30 +17,53 @@
 
 from util.context import Context
 
+LOG_THROTTLING_LIMIT_PER_CALLER = 10
+
+log_call_count = dict()
+
+
+def log_multiline_message(message: Text, caller: Text, count_for_throttling: bool = True):
+
+    # display logs called from one spot no more than the specified amount of times,
+    # count_for_throttling means we don't add to the counter but can still throttle if it already exceeds the limit
+    if count_for_throttling is True:
```

Comment:
Looking at this now, I am even more convinced that this logic would be much more simpler if we don't introduce count_for_throttling :D

Reply:
Wouldn't it be the same thing but instead of count_for_throttling we would check if caller ends with "traceback" also allowing for potential overlap with a function name? This way also seems easier if someone were to add another function calling log_multiline_message multiple times

Reply:
Unless you meant to just count each call separately in the dictionary, I thought about it but it felt inefficient, maybe it is better to keep the function call clean

Reply:
I think it makes sense to append something to the caller string in the second invocation of `log_multiline_message()` but maybe something more self-descriptive like `"-second-invocation"`

Reply:
Also this would mean we would need to check if something is a second invocation when logging that the number of calls was exceeded, so we don't get the message twice.

Reply:
I don't think we need to check anything. One of our internal methods has specific logic that it calls this method twice (even though user only called "log_with_exception" once) and this is a special case and we don't expect any more methods like this in near future.
There is one other simple solution. Merge both invocations from log_with_exception to one invocation. ;)

Reply:
Merging them is the cleanest solution but it means the message and traceback need to be in the same line - I'm not sure how much of a problem that is. As for the first part, you mean that it's fine for the logs to read something like this?
![image](https://user-images.githubusercontent.com/32064437/127331721-448b51bc-dcce-4cbb-abc6-3bc27ef9f5cb.png)

Reply:
Yeah, let's merge these two invocations. These flattened tracebacks are not readable anyway so if we append them to the end of exception message, readability won't suffer substantially

Reply:
😆 yeah they are not readable. 
Just for context:
if you use multiple lines with \n in cloudwatch (normal python behaviour) then you have multiple cloudwatch events, one for each line
ir you substitute \n to \r, then in cloudwatch you have all good in one event but locally you have mess.
Flattening to one line with some separator (what we do) is a (arguably not best) compromise :)

Reply:
With current push log_multiline_message is called only once in log_with_exception